scrapy/middleware.py

```diff
@@ -16,7 +17,7 @@ class MiddlewareManager:
 
     def __init__(self, *middlewares):
         self.middlewares = middlewares
-        self.methods = defaultdict(deque)
+        self.methods: dict[str, deque[Callable]] = defaultdict(deque)
```

Comment:
I'm surprised this syntax works (`dict[str, ...]`), it is supposed to be Python 3.9+ feature, or Python 3.7+ with `from future import __annotations__` import

Reply:
It works on 3.6 and even catches errors :)

Reply:
That's really strange!

```
>>> def foo(x: dict[str, str]): pass
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'type' object is not subscriptable
```

Reply:
Nice, new PyCharm flags my code and proposes a replacement from `dict` to `typing.Dict`, is this the correct fix?